bpf/bpf_host.c

```diff
@@ -1270,10 +1238,20 @@ int to_netdev(struct __ctx_buff *ctx __maybe_unused)
 	if (IS_ERR(ret))
 		return send_drop_notify_error(ctx, srcID, ret, CTX_ACT_DROP,
 					      METRIC_EGRESS);
-#else
-	ret = CTX_ACT_OK;
 #endif /* ENABLE_HOST_FIREWALL */
 
+#if defined(ENABLE_NODEPORT) && \
+	(!defined(ENABLE_DSR) || \
+	 (defined(ENABLE_DSR) && defined(ENABLE_DSR_HYBRID)))
+	if ((ctx->mark & MARK_MAGIC_SNAT_DONE) != MARK_MAGIC_SNAT_DONE) {
+		ret = nodeport_nat_fwd(ctx, false);
```

Comment:
In this case, the host fw will see the pod IP as src, as BPF masquerade can happen in `nodeport_nat_fwd`.